ethereum/spec/src/main/java/tech/pegasys/teku/spec/logic/versions/fulu/helpers/MiscHelpersFulu.java

```diff
@@ -125,9 +128,21 @@ public int getMaxBlobsPerBlock(final UInt64 epoch) {
         blobSchedule.stream()
             .filter(blobSchedule -> blobSchedule.epoch().isLessThanOrEqualTo(epoch))
             .max(Comparator.comparing(BlobSchedule::epoch));
-    return maybeSchedule.isPresent()
-        ? maybeSchedule.get().maxBlobsPerBlock()
-        : blobSchedule.getFirst().maxBlobsPerBlock();
+
+    final int maxBlobs =
+        maybeSchedule
+            .map(BlobSchedule::maxBlobsPerBlock)
+            .orElseGet(() -> blobSchedule.getFirst().maxBlobsPerBlock());
+    LOG.debug("Max blobs at epoch {} found to be {}", epoch, maxBlobs);
+    return maxBlobs;
+  }
+
+  public int getHighestMaxBlobsPerBlockFromSchedule() {
+    checkArgument(!blobSchedule.isEmpty(), "Blob schedules not correctly defined.");
+    return blobSchedule.stream()
```

Comment:
why to sort it each time? I think we'd better expect it sorted

Reply:
we're not sorting, we're range limiting

Reply:
as an aside we do a single sort during type initialization setting blobSchedule